randovania/data/json_data/prime3.txt

```diff
@@ -2067,59 +2427,86 @@ Hall of Remembrance
 Asset id: 6915612318130423578
 > Dock to Imperial Crypt (Upper); Heals? False
   * Normal Door to Imperial Crypt/Dock to Hall of Remembrance (Upper)
-  > Pickup (Ship Missile Expansion)
-      Use Screw Attack (Space Jump)
+  > Locked Door Ledge
+      Trivial
 
 > Dock to Imperial Crypt (Lower); Heals? False
   * Normal Door to Imperial Crypt/Dock to Hall of Remembrance (Lower)
+  > Dock to Tower (Lower)
+      Use Screw Attack (Space Jump)
   > Pickup (Screw Attack)
       Trivial
 
 > Dock to Tower (Lower); Heals? False
   * Normal Door to Tower/Dock to Hall of Remembrance (Upper)
-  > Pickup (Screw Attack)
+  > Dock to Imperial Crypt (Lower)
       Use Screw Attack (Space Jump)
 
 > Dock to Tower (Upper); Heals? False
   * Normal Door to Tower/Dock to Hall of Remembrance (Lower)
+  > Dock to Tower (Lower)
+      Trivial
   > Pickup (Ship Missile Expansion)
       Use Screw Attack (Space Jump)
+  > Locked Door Ledge
+      Use Screw Attack (Space Jump)
 
 > Pickup (Screw Attack); Heals? False
   * Pickup 48; Major Location? True
   > Dock to Imperial Crypt (Lower)
       Trivial
-  > Dock to Tower (Lower)
-      Use Screw Attack (Space Jump)
 
 > Pickup (Ship Missile Expansion); Heals? False
   * Pickup 49; Major Location? False
+  > Dock to Tower (Upper)
+      Trivial
+
+> Event - Unlock Door; Heals? False
+  > Locked Door Ledge
+      Trivial
+
+> Locked Door Ledge; Heals? False
   > Dock to Imperial Crypt (Upper)
-      Use Screw Attack (Space Jump)
+      After Hall of Remembrance Unlock Door
+  > Dock to Imperial Crypt (Lower)
+      Trivial
   > Dock to Tower (Upper)
       Use Screw Attack (Space Jump)
-  > Pickup (Screw Attack)
-      Trivial
+  > Event - Unlock Door
+      Grapple Lasso
 
 ----------------
 Imperial Crypt
 Asset id: 17976174099400013428
 > Dock to Hall of Remembrance (Upper); Heals? False
   * Normal Door to Hall of Remembrance/Dock to Imperial Crypt (Upper)
   > Dock to Imperial Caverns
-      Use Screw Attack (Space Jump)
+      Any of the following:
+          Use Screw Attack (Space Jump)
+          Boost Ball and After Imperial Crypt Ice Floor
+  > Event - Ice Floor
+      Plasma Beam
 
 > Dock to Imperial Caverns; Heals? False
   * Normal Door to Imperial Caverns/Dock to Imperial Crypt
   > Dock to Hall of Remembrance (Upper)
-      Use Screw Attack (Space Jump)
+      After Hall of Remembrance Unlock Door and Use Screw Attack (Space Jump)
   > Dock to Hall of Remembrance (Lower)
+      After Imperial Crypt Ice Floor
+  > Event - Ice Floor
       Plasma Beam
 
 > Dock to Hall of Remembrance (Lower); Heals? False
   * Normal Door to Hall of Remembrance/Dock to Imperial Crypt (Lower)
   > Dock to Imperial Caverns
-      Boost Ball
+      Boost Ball and After Imperial Crypt Ice Floor
+
+> Event - Ice Floor; Heals? False
+  * Event Imperial Crypt Ice Floor
+  > Dock to Hall of Remembrance (Upper)
+      Trivial
+  > Dock to Imperial Caverns
+      Trivial
 
 ----------------
 Tower
```

Comment:
- Upper door should connect to the lower door as "Trivial"
- Pickup should only connect to upper door